http_parser.c

```diff
@@ -2134,6 +2134,19 @@ http_parser_init (http_parser *parser, enum http_parser_type t)
   parser->http_errno = HPE_OK;
 }
 
+void
+http_parser_settings_init(http_parser_settings *settings)
+{
+  settings->on_message_begin = 0;
```

Comment:
Yeah, `memset` will be more robust if members are changed / added to the
struct.

On Sunday, January 25, 2015, Fedor Indutny <notifications@github.com> wrote:

> In http_parser.c
> <https://github.com/joyent/http-parser/pull/215#discussion_r23503445>:
>
> > @@ -2134,6 +2134,19 @@ http_parser_init (http_parser *parser, enum http_parser_type t)
> >    parser->http_errno = HPE_OK;
> >  }
> >
> > +void
> > +http_parser_settings_init(http_parser_settings *settings)
> > +{
> > +  settings->on_message_begin = 0;
>
> Why not just memset()? ;)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joyent/http-parser/pull/215/files#r23503445>.
>